src/NuGet.Clients/NuGet.Credentials/SecurePluginCredentialProvider.cs

```diff
@@ -80,51 +80,56 @@ public SecurePluginCredentialProvider(IPluginManager pluginManager, PluginDiscov
                 return taskResponse;
             }
 
-            var plugin = await _pluginManager.CreateSourceAgnosticPluginAsync(_discoveredPlugin, cancellationToken);
+            var creationResult = await _pluginManager.TryGetSourceAgnosticPluginAsync(_discoveredPlugin, OperationClaim.Authentication, cancellationToken);
 
-            if (!string.IsNullOrEmpty(plugin.Message))
+            if (creationResult.Item1)
```

Comment:
comment on what's Item1 and Item2 would be appreciated

Reply:
These are the situations in which I wish our analyzers didn't scream for var vs strong type lol.